pages/Receive.qml

```diff
@@ -75,14 +75,182 @@ Rectangle {
         spacing: 20
 
         ColumnLayout {
-            id: addressRow
-            spacing: 0
+            id: selectedAddressDetailsColumn
+            Layout.alignment: Qt.AlignHCenter
+            spacing: 11
+            property int qrSize: 220
 
-            MoneroComponents.LabelSubheader {
+            Rectangle {
+                id: qrContainer
+                color: MoneroComponents.Style.blackTheme ? "white" : "transparent"
                 Layout.fillWidth: true
-                fontSize: 24
+                Layout.alignment: Qt.AlignHCenter
+                Layout.maximumWidth: parent.qrSize
+                Layout.preferredHeight: width
+                radius: 4
+
+                Image {
+                    id: qrCode
+                    anchors.fill: parent
+                    anchors.margins: 1
+                    smooth: false
+                    fillMode: Image.PreserveAspectFit
+                    source: "image://qrcode/" + TxUtils.makeQRCodeString(appWindow.current_address)
+
+                    MouseArea {
+                        anchors.fill: parent
+                        hoverEnabled: true
+                        cursorShape: Qt.PointingHandCursor
+                        acceptedButtons: Qt.LeftButton | Qt.RightButton
+                        onClicked: {
+                            if (mouse.button == Qt.LeftButton){
+                                selectedAddressDetailsColumn.qrSize == 220 ? selectedAddressDetailsColumn.qrSize = 300 : selectedAddressDetailsColumn.qrSize = 220;
+                            } else if (mouse.button == Qt.RightButton){
+                                qrMenu.x = this.mouseX;
+                                qrMenu.y = this.mouseY;
+                                qrMenu.open()
+                            }
+                        }
+                    }
+                }
+
+                Menu {
+                    id: qrMenu
+                    title: "QrCode"
+
+                    MenuItem {
+                        text: qsTr("Save as Image") + translationManager.emptyString;
+                        onTriggered: qrFileDialog.open()
+                    }
+                }
+            }
+
+            MoneroComponents.TextPlain {
+                id: selectedaddressIndex
+                Layout.alignment: Qt.AlignHCenter
+                Layout.maximumWidth: 220
```

Comment:
Can you add a Layout.topMargin here? The margin between QR code and address isn't the same as between address and description.